lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+<template>
+	<div
+		class="modal"
+		:class="{
+			'-danger': danger,
+		}"
+		@click.stop="$emit('close-modal')"
+	>
+		<div class="modal__wrapper" :class="{ '-small': size === MODAL_SIZES.SMALL }">
+			<div
+				v-if="headerImage"
+				class="modal__image"
+				:style="{ backgroundImage: 'url(' + headerImage + ')' }"
+			></div>
+			<div class="modal__content" :class="{ '-centered': contentCentered }">
+				<div class="modal__header">
+					<wnl-icon-button
+						touchable
+						:icon="ICONS.FA_XMARK"
+						class="modal__close"
+						:size="ICON_SIZES.SMALL"
+						:elevation="BUTTON_ELEVATIONS.X_SMALL"
+						:color="ICON_BUTTON_COLORS.NEUTRAL_WEAK"
+						@click.native.stop="$emit('close-modal')"
+					/>
+					<feature-icon
+						v-if="headerFeatureIcon"
+						class="modal__headerFeatureIcon"
+						:color="calcHeaderFeatureIconColor"
+						:icon="headerFeatureIcon"
+						:size="FEATURE_ICON_SIZES.X_LARGE"
+						:double-background="true"
+					/>
+					<h3
+						class="modal__headerTitle"
+						:class="{ '-small': headerTitleSize === MODAL_HEADER_TITLE_SIZES.SMALL }"
+						>{{ headerTitle }}</h3
+					>
+					<h6 v-if="headerSubtitle" class="modal__headerSubtitle">{{
+						headerSubtitle
+					}}</h6>
+				</div>
+				<div v-if="$slots.default" class="modal__textContent">
+					<slot />
+				</div>
+				<div class="modal__footer" :class="{ '-singleColumn': calcSingleColumn }">
+					<div
+						v-if="footerTertiaryButtonText || footerCheckboxText"
+						class="modal__footerColumn --ctaSecondary"
+					>
+						<div v-if="footerCheckboxText" ref="button" class="modal__checkbox">
+							<input
+								id="modal__checkboxInput"
+								type="checkbox"
+								:checked="false"
+								class="modal__checkboxInput"
+								@change="$emit('checkbox-change')"
+							/>
+							<label for="modal__checkboxInput" class="modal__checkboxLabel">
+								{{ footerCheckboxText }}
+							</label>
+						</div>
+						<wnl-button
+							v-if="footerTertiaryButtonText"
+							ref="button"
+							class="modal__buttonSecondary"
+							:type="BUTTON_TYPES.TEXT"
+							:color="BUTTON_COLORS.NEUTRAL"
+							:icon-left="ICONS[footerTertiaryButtonIcon]"
```

Comment:
in props validator, you assume `footerTertiaryButtonIcon` is a valid Icon object, not icon name. IMO, we shouldn't use `ICONS` here. The same for other buttons

Reply:
[`16ed1c0` (#248)](https://github.com/bethinkpl/design-system/pull/248/commits/16ed1c00e2363f0cceb80fbaee2fe1452d89de1e)